akka-actor/src/main/scala/akka/japi/function/Function.scala

```diff
@@ -23,6 +31,16 @@ trait Function2[-T1, -T2, +R] extends java.io.Serializable {
   def apply(arg1: T1, arg2: T2): R
 }
 
+object Function2 {
+  final class Wrapper[T1, T2, R](f: (T1, T2) ⇒ R)
+    extends ((T1, T2) ⇒ R) with Function2[T1, T2, R] {
+    def apply(arg1: T1, arg2: T2): R = f(arg1, arg2)
+  }
+
+  def apply[T1, T2, R](f: (T1, T2) ⇒ R): Function2[T1, T2, R] =
+    new Wrapper[T1, T2, R](f)
+}
+
```

Comment:
I still feel that this and the changes to Keep should not be in this PR if we are to merge it.